location/src/main/java/org/odk/collect/location/satellites/GpsStatusSatelliteInfoClient.kt

```diff
@@ -35,3 +43,30 @@ private class GpsStatusSatellitesLiveData(private val locationManager: LocationM
         }
     }
 }
+
+@SuppressLint("MissingPermission")
+@RequiresApi(Build.VERSION_CODES.N)
+private class GnssStatusSatellitesLiveData(private val locationManager: LocationManager) :
+    NonNullLiveData<Int>(0) {
+
+    private val gnssStatusCallback = object : GnssStatus.Callback() {
+        override fun onSatelliteStatusChanged(status: GnssStatus) {
+            super.onSatelliteStatusChanged(status)
+            var usedInFix = 0
```

Comment:
Not essential (and mostly subjective), but you can avoid needing any mutation here with a `fold`:

```kotlin
value = (0 until status.satelliteCount).fold(0) { count, index ->
    when {
        status.usedInFix(index) -> count + 1
       else -> count
    }
}
```

Reply:
Done.